api/nativeMessaging/main.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+let port = null;
+
+const appendMessage = (text) => {
+  document.getElementById('response').innerHTML += '<p>' + text + '</p>';
+}
+
+const updateUiState = () => {
+  if (port) {
+    document.getElementById('connect-button').style.display = 'none';
+    document.getElementById('input-text').style.display = 'block';
+    document.getElementById('send-message-button').style.display = 'block';
+  } else {
+    document.getElementById('connect-button').style.display = 'block';
+    document.getElementById('input-text').style.display = 'none';
+    document.getElementById('send-message-button').style.display = 'none';
+    document.getElementById('response').innerHTML = '';
+  }
+}
+
+const sendNativeMessage = () => {
+  try {
+    const message = JSON.parse(document.getElementById('input-text').value);
+    port.postMessage(message);
+    appendMessage('Sent message: <b>' + JSON.stringify(message) + '</b>');
+  } catch (e) {
+    console.error(e.message);
+    if (e.message === 'Attempting to use a disconnected port object') {
```

Comment:
is there actually a better class to check for here (e.g. "x instanceof SomeError")? Checking the exact string is kinda gross.

Reply:
That is a matter of style, not necessarily related to the Native Messaging. I prefer to get the literal message.